src/code/z_bgcheck.c

```diff
@@ -4210,126 +4212,85 @@ u32 func_800C99FC(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
- */
-u32 func_800C9A24(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+BgWallType SurfaceType_GetWallType(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     return SurfaceType_GetData(colCtx, poly, bgId, 0) >> 21 & 0x1F;
 }
 
-/**
- * SurfaceType Get Wall Flags
- */
-s32 func_800C9A4C(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
-    return sWallFlags[func_800C9A24(colCtx, poly, bgId)];
+s32 SurfaceType_GetWallFlags(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+    return sWallFlags[SurfaceType_GetWallType(colCtx, poly, bgId)];
 }
 
-/**
- * SurfaceType Is Wall Flag (1 << 0) Set
- */
-s32 func_800C9A7C(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
-    return (func_800C9A4C(colCtx, poly, bgId) & 1) ? true : false;
+s32 SurfaceType_CheckWallFlag0(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+    return (SurfaceType_GetWallFlags(colCtx, poly, bgId) & WALL_FLAG_0) ? true : false;
 }
 
-/**
- * SurfaceType Is Wall Flag (1 << 1) Set
- */
-s32 func_800C9AB0(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
-    return (func_800C9A4C(colCtx, poly, bgId) & 2) ? true : false;
+s32 SurfaceType_CheckWallFlag1(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+    return (SurfaceType_GetWallFlags(colCtx, poly, bgId) & WALL_FLAG_1) ? true : false;
 }
 
-/**
- * SurfaceType Is Wall Flag (1 << 2) Set
- */
-s32 func_800C9AE4(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
-    return (func_800C9A4C(colCtx, poly, bgId) & 4) ? true : false;
+s32 SurfaceType_CheckWallFlag2(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+    return (SurfaceType_GetWallFlags(colCtx, poly, bgId) & WALL_FLAG_2) ? true : false;
 }
 
-u32 func_800C9B18(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+BgFloorProperty SurfaceType_GetFloorProperty2(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     return SurfaceType_GetData(colCtx, poly, bgId, 0) >> 26 & 0xF;
 }
 
-/**
- * SurfaceType Get Floor Property
- */
-u32 func_800C9B40(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+BgFloorProperty SurfaceType_GetFloorProperty(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     return SurfaceType_GetData(colCtx, poly, bgId, 0) >> 26 & 0xF;
 }
 
-/**
- * SurfaceType Is Floor Minus 1
- */
-u32 func_800C9B68(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+u32 SurfaceType_IsSoft(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     return SurfaceType_GetData(colCtx, poly, bgId, 0) >> 30 & 1;
 }
 
-/**
- * SurfaceType Is Horse Blocked
- */
 u32 SurfaceType_IsHorseBlocked(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
     return SurfaceType_GetData(colCtx, poly, bgId, 0) >> 31 & 1;
 }
 
-u32 func_800C9BB8(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+BgSurfaceSfxType SurfaceType_GetSfxType(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
```

Comment:
I don't understand why Dragorn or you have decided to name this after sound effects; this is used to return the material that the surface is made out of, of which `SurfaceType_GetSfxId` calls this to determine which sound effect to play to match the surface being walked on.

You can see that many actors call `func_800C9BB8` to determine what particle effects to generate when interacting with different surfaces. `en_Dodongo` checks this property to see if it's walking on snow in order to generate white particles instead of dirt colored ones. `en_Tite` checks if the surface it has landed on is ground/sand to determine if it should generate dust particles, snow if it should generate show particles.

Reply:
Because I was mistaken and missed the few uses that weren't sfx-related in OoT

In OoT `SurfaceType_GetSfxType` is used directly only twice outside bgcheck, in z_player, partly for sfx stuff at both spots

also `Player.unk_89E` usage indicates there's conflating between returns of `SurfaceType_GetSfxType` and `SurfaceType_GetSfxId` which I have yet to understand what's going on

And `SurfaceType_GetSfxId` is overwhelmingly used for getting a sfxId to play it, there's just this weird one in `z_en_js`:
```c
if (SurfaceType_GetSfxId(&play->colCtx, this->actor.floorPoly, this->actor.floorBgId) ==
    (NA_SE_PL_WALK_SAND - SFX_FLAG)) {
    Math_ApproachF(&this->actor.shape.yOffset, sREG(80) + -2000.0f, 1.0f, (sREG(81) / 10.0f) + 50.0f);
}
```

[In OoT] We are talking about two functions which are referenced a grand total of 13 times in OoT, with 10 of those uses being strictly about sfx, and the other 3 involving sfx but also other stuff

So I think it's pretty harsh to say you can't see why I went with "sfx" for naming these in OoT

I'm open to hearing your suggestions

Reply:
The mixed usage of "`SurfaceType_GetSfxType `" and `SurfaceType_GetSfxId` makes me question whether `SurfaceType_GetSfxId` is in fact actually returning a sfxId. We currently express it as returning a SfxId, but the adding and subtracting of `SFX_FLAG` in many instances around where the function is being called tells me at the very least that something is being represented wrong; I'm not sure if it's just that `NA_SE_PL_WALK` sfx shouldn't have SFX_FLAG pre-added, or if there is meant to be a different enumeration altogether that is translated into a sfxId. 

At the same time, the new MM function `func_800C9C24` is used to determine if footprints should be imprinted on the ground, and follows the same pattern as `SurfaceType_GetSfxId` which makes me think we still got it right.

Reply:
> but the adding and subtracting of `SFX_FLAG` in many instances around where the function is being called tells me at the very least that something is being represented wrong

What do you mean? Every time this function is called they add the `SFX_FLAG`, they don't do an extra subtraction.
Actually iirc there's nothing in the code that points subtracting a `SFX_FLAG` is the correct way to handle it, it always is being added, only subtracted on constant-folded expressions, which point at our macros being "wrong" (because they have the `SFX_FLAG` pre-added)

We probably need some info from our audio guy @engineer124 

Personally I think both names are fine for describing what the functions are doing

Reply:
`SFX_FLAG` is just a flag to characterize whether the sfx is played for its full length or for 1 frame (and requires repeated calls). The sfxId with or without the `SFX_FLAG` is still a valid sfxId (with the sole exception of 0, but that might be a bug on the audio side of things).

i.e. for `NA_SE_PL_JUMP` as an example, `0x810` and `0x10` are both valid `sfxId`'s that will play this sfx. Except `sfxId = 0x10` means you need to call the sfx every frame to continue playing, while `0x810` means to play the full sfx sound with only 1 function call.

It is a conscious choice to make the sfxId defines with the flag already set, because we don't want `+ SFX_FLAG` everywhere in the repo to toggle this flag on. Instead, we chose to have `- SFX_FLAG` to toggle the flag off as this shows up much less frequently in the repo.

All that's happening with these temps related to `SurfaceType_GetSfxId` is that `SfxId` is having it's `SFX_FLAG` toggled on and off.

The one instance of `SurfaceType_GetSfx` being used as a SfxId is a very strange case and is likely a mistake from the devs imo.